### PR TITLE
Fix copy job listing bug

### PR DESCRIPTION
**Pull request by dushan-madetech**
## Trello card URL:

https://trello.com/c/IQmSekv5

## Changes in this PR:

- Add test to check for vacancy fields not being copied e.g. page views
- Add fix for copying vacancy bug

### CODE REVIEW

File: spec/services/copy_vacancy_spec.rb [hunk 1]
```diff
@@ -26,5 +26,28 @@
...
+                         slug: 'maths-teacher',
+                         weekly_pageviews: 4,
+                         total_pageviews: 4,
+                         weekly_pageviews_updated_at: Time.zone.today - 5.days,
+                         total_pageviews_updated_at: Time.zone.today - 5.days,
+                         total_get_more_info_clicks: 6,
+                         total_get_more_info_clicks_updated_at: Time.zone.today - 5.days)
+
+        result = described_class.new(vacancy).call
+        expect(Vacancy.find(result.id).slug).to_not eq('maths_teacher')
```
Comment (adrianclay) [inline spec/services/copy_vacancy_spec.rb]: The slug above has a dash?

File: spec/services/copy_vacancy_spec.rb [hunk 2]
```diff
@@ -26,5 +26,28 @@
...
+        expect(Vacancy.find(result.id).weekly_pageviews).to eq(0)
+        expect(Vacancy.find(result.id).weekly_pageviews_updated_at).to_not eq(Time.zone.today - 5.days)
```
Comment (adrianclay) [inline spec/services/copy_vacancy_spec.rb]: What do we want the `updated_at` to be? Maybe investigate how the field is used, then make a call on what a sensible value is?

File: spec/services/copy_vacancy_spec.rb [hunk 3]
```diff
@@ -26,5 +26,28 @@
 
       Timecop.return
     end
+
+    context 'not all fields are copied' do
+      it 'should not copy fields that should be unique' do
```
Comment (adrianclay) [inline spec/services/copy_vacancy_spec.rb]: I might separate out the slug part and the counters into separate `it` blocks.  They feel like separate enough tests, with different reasons to fail.

File: spec/services/copy_vacancy_spec.rb (hunk 1 above)
Comment (dushan-madetech) [reply spec/services/copy_vacancy_spec.rb]: Fixed in https://github.com/DFE-Digital/teacher-vacancy-service/pull/942/commits/39703520053b37b6e1a9095598d71df36b0bfd75

File: spec/services/copy_vacancy_spec.rb (hunk 2 above)
Comment (dushan-madetech) [reply spec/services/copy_vacancy_spec.rb]: Set to current time using Time.zone.now, see https://github.com/DFE-Digital/teacher-vacancy-service/pull/942/commits/39703520053b37b6e1a9095598d71df36b0bfd75 for this change

File: spec/services/copy_vacancy_spec.rb (hunk 3 above)
Comment (dushan-madetech) [reply spec/services/copy_vacancy_spec.rb]: Separated the test into multiple individual tests, see https://github.com/DFE-Digital/teacher-vacancy-service/pull/942/commits/39703520053b37b6e1a9095598d71df36b0bfd75 for this change

**Pull request merged**